### PR TITLE
Add AWS auth info to upgrade guide

### PR DESCRIPTION
**Pull request by tyrannosaurus-becks**
This PR adds a note to the 1.4.0 upgrade guide regarding AWS auth metadata and how it may effect performance.

### CODE REVIEW

File: website/pages/docs/upgrading/upgrade-to-1.4.0.mdx [hunk 1]
```diff
@@ -7,4 +7,24 @@ description: |-
...
+1.4.0 we began adding metadata to tokens issued with this method. While the
+metadata does help with tying tokens to a particular person or machine, it
+also can also take a performance toll.
+
+The specific fields added for `iam` tokens were `client_arn`, `canonical_arn`,
+`client_user_id`, `auth_type`, `inferred_entity_type`, `inferred_entity_id`,
+`inferred_aws_region`, and `account_id`. The specific fields added for `ec2`
+logins were `instance_id`, `region`, `account_id`, and `ami_id`.
+
+Companies who may experience a high rate of change in these fields for a role
```
Comment (mladlow) [inline website/pages/docs/upgrading/upgrade-to-1.4.0.mdx:27]: Companies -> users, maybe? Companies seems like a word we don't use often in our docs, and you use 'users' earlier in this section.

Do we have an example of the type of situation where you might see a high rate of change for these fields? I can't visualize this, but I'm also probably not really the intended audience.

Comment (tyrannosaurus-becks) [reply website/pages/docs/upgrading/upgrade-to-1.4.0.mdx]: Hi! Thanks for looking at this.

I tried elaborating more about who would see a high rate of change [here](https://github.com/hashicorp/vault/pull/8731/files#diff-3772dd3d3374cb57b6e5e8656299772aR8) and down. What do you think?

File: website/pages/docs/upgrading/upgrade-to-1.3.2.mdx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  This page contains the list of deprecations and important or breaking changes
+  for Vault 1.3.2. Please read it carefully.
+---
+
+# Overview
+
+This page contains the list of deprecations and important or breaking changes
+for Vault 1.3.0 compared to 1.3.2. Please read it carefully.
+
+@include 'partials/aws-auth-metadata-issue.mdx'
```
Comment (mladlow) [inline website/pages/docs/upgrading/upgrade-to-1.3.2.mdx:15]: Oh, neat!

File: website/pages/partials/aws-auth-metadata-issue.mdx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+- `inferred_aws_region`
+- `account_id`
+
+Metadata fields for `ec2` roles:
+- `instance_id`
+- `region`
+- `account_id`
+- `ami_id`
+
+For example, if you use `role_id` for identity and only `iam` roles, and
```
Comment (mladlow) [inline website/pages/partials/aws-auth-metadata-issue.mdx:32]: This is a great example!

Comment (tyrannosaurus-becks) [reply website/pages/partials/aws-auth-metadata-issue.mdx:32]: Thanks!

**Pull request merged**